purchase_allowed_product/models/product.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# © 2017 Today Mourad EL HADJ MIMOUNE @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.model
+    def search(self, args, offset=0, limit=None, order=None, count=False):
+        restrict_supplier_id = self.env.context.get(
+            'restrict_supplier_id', False)
+        use_only_supplied_product = self.env.context.get(
+            'use_only_supplied_product', False)
```

Comment:
it seems second arg False is not mandatory here and above because it's the default one